server/src/com/cloud/storage/snapshot/SnapshotManagerImpl.java

```diff
@@ -1331,6 +1331,21 @@ public boolean canOperateOnVolume(Volume volume) {
         return true;
     }
 
+    @Override
+    public void cleanupSnapshotsByVolume(Long volumeId) {
+        List<SnapshotInfo> infos = snapshotFactory.getSnapshots(volumeId, DataStoreRole.Primary);
+        for(SnapshotInfo info: infos) {
+            try {
+               if(info != null) {
```

Comment:
Do you need this null check?
This only happen if one uses the `set` method of List object to replace a position with a `Null` value